python/lib/database.py

```diff
@@ -0,0 +1,593 @@
+import csv
+import psycopg2
+import subprocess
+from psycopg2.extras import RealDictCursor
+from lib.sqlprettyprint import sql_print, bash_print
+
+
+def get_db_handle():
+    return psycopg2.connect(
+        dbname="visionzero",
+        user="vz",
+        password="vz",
+        host="db",
+        cursor_factory=RealDictCursor,
+    )
+
+
+def ensure_extensions_exists(db):
+    with db.cursor() as cursor:
+        # Create postgis extension if it does not exist
+        sql = "CREATE EXTENSION IF NOT EXISTS postgis;"
+        sql_print(sql)
+        cursor.execute(sql)
+
+        # Create periods extension if it does not exist
+        sql = "CREATE EXTENSION IF NOT EXISTS periods CASCADE;"
+        sql_print(sql)
+        cursor.execute(sql)
```

Comment:
I believe this is a widely used extension, and it aims to add temporal table support as specified in the 2016 SQL standard which was brought forward from the 2011 standard. The authors aim to implement this support in such a way that they can hand over the support to core postgres when the PG team implements the new SQL standards. I guess if you think government moves slow, check out database standards.. :ice_cube: 

But either way, it's a silver bullet to the temporal tacking problem and (future) standards compliant to boot.